consensus/src/chained_bft/network.rs

```diff
@@ -473,15 +639,8 @@ impl<T: Payload> NetworkTask<T> {
         peer_id: AccountAddress,
     ) -> anyhow::Result<()> {
         let sync_info = SyncInfo::try_from(sync_info)?;
-        match sync_info.epoch().cmp(&self.epoch()) {
```

Comment:
Perhaps we could start with a smaller commit that just unifies the common logic on reviewing the messages epoch against the current epoch? There looks like a lot of copied code that could be isolated into a common function:

`
  fn check_epoch(&self, epoch: u64) {
      ...
   }
`

Reply:
Yes this additional change could be made in a separate commit indeed.